src/main/java/com/aws/greengrass/logmanager/LogManagerService.java

```diff
@@ -658,18 +654,27 @@ private void processLogsAndUpload() throws InterruptedException {
                     Collectors.toList());
 
             if (unitsOfWork.isEmpty()) {
-                TimeUnit.SECONDS.sleep(periodicUpdateIntervalSec);
                 isCurrentlyUploading.set(false);
             } else {
                 unitsOfWork.forEach((unit) -> {
                     CloudWatchAttempt cloudWatchAttempt = logsProcessor.processLogFiles(unit);
                     uploader.upload(cloudWatchAttempt, 1);
                 });
+                emitEventStatus(EventType.LOG_GROUP_PROCESSED);
             }
+            // after handle one cycle, we sleep for interval to avoid seamless scanning and processing next cycle.
+            TimeUnit.SECONDS.sleep(periodicUpdateIntervalSec);
```

Comment:
⚠️ This might not be what we want - it concerns me a bit that tests pass with this (meaning we don't have a test for this) - This forces the thread to always sleep after each cycle if a client sets it to 20 minutes then we will still have a similar effect to the one we have today where active log files take forever to be uploaded.